src/Wordpress/CustomPostType/Timeframe.php

```diff
@@ -1068,11 +1068,30 @@ public static function removeIrrelevantPostmeta( \CommonsBooking\Model\Timeframe
 			\CommonsBooking\Model\Timeframe::META_CREATE_BOOKING_CODES,
 			\CommonsBooking\Model\Timeframe::META_SHOW_BOOKING_CODES,
 		];
+		//remove multi-select postmeta if not relevant (#507)
+		$onlyRelevantForHolidays = [
+			\CommonsBooking\Model\Timeframe::META_ITEM_IDS,
+			\CommonsBooking\Model\Timeframe::META_LOCATION_IDS,
+			\CommonsBooking\Model\Timeframe::META_ITEM_CATEGORY_IDS,
+			\CommonsBooking\Model\Timeframe::META_LOCATION_CATEGORY_IDS,
+			\CommonsBooking\Model\Timeframe::META_ITEM_SELECTION_TYPE,
+			\CommonsBooking\Model\Timeframe::META_LOCATION_SELECTION_TYPE,
+		];
+
 		if ($timeframe->getType() != Timeframe::BOOKABLE_ID) {
 			foreach ( $onlyRelevantForBookable as $metaKey ) {
 				delete_post_meta( $timeframe->ID, $metaKey );
 			}
 		}
+
+		if ($timeframe->getType() != Timeframe::HOLIDAYS_ID) {
+			foreach ( $onlyRelevantForHolidays as $metaKey ) {
+				delete_post_meta( $timeframe->ID, $metaKey );
+			}
+			//reset to manual selection
```

Comment:
Eine andere noch nicht-ausgereifte Idee (fürs Backlog) wäre die Holiday-Funktionalitäten in der Timeframe-Klasse in eine eigene (davon abgeleitete) Holiday-Timeframe-Klasse auszulagern.

Dann könnten so Zeilen wie hier (welche sich z.B. in der Migration wiederholen) in der HolidayTimeframe-Klasse liegen.

Reply:
@hansmorb Dafür spricht mMn auch, das wir bei manchen Methoden per Dokumentation empfehlen, diese nicht für Holidy-Timeframes zu benutzen. Das könnte man via Code auch schön unterteilen und so klarer definieren.

Reply:
Oder vielleicht auch getItems() für sämtliche TFs zu nutzen, das entspricht ja auch unseren Plänen TFs unabhängig von Holidays mehreren Artikeln zuweisen zu können.